path_service_principal_test.go

```diff
@@ -764,23 +767,23 @@ func TestRoleAssignmentWALRollback(t *testing.T) {
 		provider := client.provider.(*provider)
 		spObjID := findServicePrincipalID(t, provider.spClient, appID)
 
-		assertServicePrincipalExists(t, provider.spClient, spObjID)
+		assertServicePrincipalExistence(t, provider.spClient, spObjID, true)
 
 		// Verify that the role assignments were created. Get the assignment
 		// info from Azure and verify it matches the Reader role.
 		raIDs := credsResp.Secret.InternalData["role_assignment_ids"].([]string)
 		equal(t, 2, len(raIDs))
 
-		ra, err := provider.raClient.GetByID(context.Background(), raIDs[0])
+		ra, err := provider.raClient.GetByID(context.Background(), raIDs[0], nil)
 		assertErrorIsNil(t, err)
 
-		roleDefs, err := provider.ListRoleDefinitions(context.Background(), fmt.Sprintf("subscriptions/%s", subscriptionID), "")
+		roleDefs, err := provider.ListRoleDefinitions(nil, fmt.Sprintf("subscriptions/%s", subscriptionID), "")
```

Comment:
This test is failing with `unexpected error: net/http: nil Context`
```suggestion
		roleDefs, err := provider.ListRoleDefinitions(context.Background(), fmt.Sprintf("subscriptions/%s", subscriptionID), "")
```

Reply:
Aah I thought I got all of them 😓 Thanks!